Posts/2021/04/tfl-IsUserALocalAdministrator.md

```diff
@@ -0,0 +1,130 @@
+Cmdlet          New-LocalGroup             1.0.0.0    Microsoft.PowerShell.LocalAccounts
+Cmdlet          New-LocalUser              1.0.0.0    Microsoft.PowerShell.LocalAccounts
+Cmdlet          Remove-LocalGroup          1.0.0.0    Microsoft.PowerShell.LocalAccounts
+Cmdlet          Remove-LocalGroupMember    1.0.0.0    Microsoft.PowerShell.LocalAccounts
+Cmdlet          Remove-LocalUser           1.0.0.0    Microsoft.PowerShell.LocalAccounts
+Cmdlet          Rename-LocalGroup          1.0.0.0    Microsoft.PowerShell.LocalAccounts
+Cmdlet          Rename-LocalUser           1.0.0.0    Microsoft.PowerShell.LocalAccounts
+Cmdlet          Set-LocalGroup             1.0.0.0    Microsoft.PowerShell.LocalAccounts
+Cmdlet          Set-LocalUser              1.0.0.0    Microsoft.PowerShell.LocalAccounts
+```
+
+As you can tell, these cmdlets allow you to add, remove, change, enable and disable a local user or local group
+And they allow you to add, remove and get the local group's members.
+These cmdlets are broadly similar to the ActiveDirectory cmdlets, but work on local users.
+And as noted above, you can use domain users/groups as a member of a local group should you wish or need to.
+
+You use the `Get-LocalGroupMember` command to view the members of a local group, like this:
+
+```powershell-console
+PS> Get-LocalGroupMember -Group 'Administrators'
+
+ObjectClass Name                     PrincipalSource
+----------- ----                     ---------------
+Group       COOKHAM\Domain Admins    ActiveDirectory
+User        COOKHAM24\Administrator  Local
+User        COOKHAM\JerryG           ActiveDirectory
+User        COOKHAM24\Dave           Local
+```
+
+As you can see in this output, the local Administrators group on this host contains domain users and groups as well as local users
+
+## Is the User an Administrator?
+
+It's easy to get membership of any local group, as you saw above.
+But what if you want to find out if a given user is a member of some group?
+Or, determine whether the user running a given script is a member of the administrator's group?
+That too is pretty easy and take a couple of steps.
+One way you can get the name of the current user is by using `whoami.exe`.
+Then you can get the members of the Administrator's group.
+Finally, you check to see if the currently logged on user is a member of the group.
+All of which looks like this:
+
+```powershell-console
+PS> # Get who I am
+PS> $Me = whoami.exe
+PS> $Me 
+Cookham\JerryG
+
+PS> # Get members of administrators group
+PS> $Admins = Get-LocalGroupMember -Name Administrators | 
+       Select-Object -ExpandProperty name
+
+PS> # Check to see if this user is an administrator and act accordingly
+PS> if ($Admins -Contains $me) {
+      "$Me is a local administrator"} 
+    else {
+     "$Me is NOT a local administrator"}
+Cookham\JerryG is a local administrator
+```
+
+If the user running the script, then **$Me** is a user in the local administrator's group.
```

Comment:
```suggestion
If the user running the script, then **$Me** is a user in the local administrator's group.
```
```suggestion
If the user running the script, then `$Me` is a user in the local administrator's group.
```

Reply:
updated the relevant sentence in the next commit. Also fixed the casing for $ME